### PR TITLE
[Cosmos] Re-enable emulator tests for ci pipeline

### PR DESCRIPTION
**Pull request by simorenoh**
Trying to re-enable the emulator tests in our ci pipeline for reliability purposes. Will probably be making several PRs since the only way I am aware of to check these updates is to run them once live. If there's another way I can test this please let me know!

Samples changed are just based on https://github.com/Azure/azure-sdk-for-python/issues/30735 - figured it was small enough I could sneak it in here (one line change).

### CODE REVIEW

File: eng/pipelines/templates/stages/cosmos-sdk-client.yml [hunk 1]
```diff
@@ -22,36 +19,38 @@ extends:
...
+          parameters:
+            EmulatorMsiUrl: ${{ parameters.EmulatorMsiUrl }}
+            StartParameters: ${{ parameters.StartParameters }}
+
+        - template: /eng/pipelines/templates/steps/build-test.yml
+          parameters:
+            TestMarkArgument: not globaldb
+            EnvVars:
+              ACCOUNT_HOST: 'https://localhost:8081/'
+              ACCOUNT_KEY: 'C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw=='
```
[the remark this answers is not in the document]
Comment (simorenoh) [reply eng/pipelines/templates/stages/cosmos-sdk-client.yml:51]: Yeah this is the publicly documented key for the emulator: https://learn.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=ssl-netstd21#authenticate-requests

File: eng/pipelines/templates/stages/cosmos-sdk-client.yml [hunk 2]
```diff
@@ -22,36 +19,38 @@ extends:
...
+        - template: /eng/common/pipelines/templates/steps/cosmos-emulator.yml
+          parameters:
+            EmulatorMsiUrl: ${{ parameters.EmulatorMsiUrl }}
+            StartParameters: ${{ parameters.StartParameters }}
+
+        - template: /eng/pipelines/templates/steps/build-test.yml
+          parameters:
+            TestMarkArgument: not globaldb
+            EnvVars:
+              ACCOUNT_HOST: 'https://localhost:8081/'
```
Comment (scbedd) [inline eng/pipelines/templates/stages/cosmos-sdk-client.yml:50]: You might need to set a couple more so that they run in "live" mode against the local cosmos emulator right? Playback tests aren't exercising the emulator if I'm understanding what this does correctly.

**Review by benbp — APPROVED** [review]

File: eng/common/pipelines/templates/steps/cosmos-emulator.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 parameters:
   EmulatorInstallPath: "$(Agent.HomeDirectory)/../../Program Files/Azure Cosmos DB Emulator/Microsoft.Azure.Cosmos.Emulator.exe"
   EmulatorMsiUrl: "https://aka.ms/cosmosdb-emulator"
-  StartParameters: ''
+  StartParameters: '/noexplorer /noui /enablepreview /EnableSqlComputeEndpoint /SqlComputePort=9999 /disableratelimiting /partitioncount=50 /consistency=Strong'
```
Comment (scbedd) [inline eng/common/pipelines/templates/steps/cosmos-emulator.yml:4]: I think you need to revert the changes to this right here. This is coming from `eng/common` and you need to follow a different workflow to merge changes to files under that path. (PR to sdk-tools, etc)

However, the change of this default is un-necessary, you providing the `EmulatorStartParameters` will definitely be good enough to make this work.
